packages/ui5-application-writer/src/types.ts

```diff
@@ -12,39 +12,38 @@ export interface Package {
 
 export interface App {
     id: string;
-    uri?: string; // What is this?
-    version?: string; // What is this?
+    version?: string;
     title?: string;
     description?: string;
     baseComponent?: string; // UI5 Component `required` by Component.js
-    flpAppId?: string;
     startFile?: string;
     localStartFile?: string;
 }
 
+export type UI5Framework = 'SAPUI5' | 'OpenUI5';
+
 export interface UI5 {
-    framework?: 'SAPUI5' | 'OpenUI5';
-    frameworkUrl?: string;
-    minUI5Version?: string;
-    version?: string;
-    localVersion?: string;
-    typesVersion?: string;
-    descriptorVersion?: string;
-    ui5Libs?: string | string[];
-    ui5Theme?: string;
-    directLoad?: boolean;
+    framework: UI5Framework;
+    frameworkUrl: string;
+    minUI5Version: string;
+    version: string;
+    localVersion: string;
+    typesVersion: string;
+    descriptorVersion: string;
+    ui5Libs: string | string[];
+    ui5Theme: string;
 }
 
 // Additional configurable features
 export interface AppOptions {
-    codeAssist?: boolean; // Enables code assist
-    eslint?: boolean; // Enables Fiori linting
-    sapux?: boolean; // Enables Fiori Tools Suite integration
+    codeAssist: boolean; // Enables code assist
+    eslint: boolean; // Enables Fiori linting
+    sapux: boolean; // Enables Fiori Tools Suite integration
 }
 
 export interface Ui5App {
     app: App;
-    appOptions?: AppOptions;
-    ui5?: UI5;
+    appOptions?: Partial<AppOptions>;
```

Comment:
Personally I wouldn't use Partial and instead define optional properties on the type AppOptions as it was before.
Of course if currently makes no difference.

Reply:
I thought it has a nice touch using `Partial` because you see right away that every single property is optional.